test/operator-framework-e2e/read_manifests.go

```diff
@@ -0,0 +1,59 @@
+package operatore2e
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+)
+
+type Object struct {
+	Kind       string `yaml:"kind"`
+	APIVersion string `yaml:"apiVersion"`
+	Metadata   struct {
+		Name string `yaml:"name"`
+	} `yaml:"metadata"`
+}
```

Comment:
I think you are looking for something like `k8s.io/apimachinery/pkg/runtime/serializer`. Here is an example usage in our project:

https://github.com/operator-framework/operator-controller/blob/fa84c733ade0f71f33085a978ae0163f570d48b7/cmd/resolutioncli/input_manifests.go#L44-L45

Reply:
Updated the code using the universal decoder. However, in cases where a single file contains multiple YAML documents separated by `---`, the universal decoder recognizes only the first resource. This situation is relevant,  `plain+v0` bundle generated through `kustomize`, the manifest has multiple YAML documents are combined into one file using `--- `separators. Is there a decoder solution that can effectively handle scenarios involving multiple YAML documents within a single file? This is now handled by splitting the content of the YAML file and decoding each of them using the universal decoder.

Reply:
Good point!

I think `YAMLToJSONDecoder` from `k8s.io/apimachinery/pkg/util/yaml` should do the trick.

I think you should be able to get kind, api version and name if you decode into `Unstructured` from `k8s.io/apimachinery/pkg/apis/meta/v1/unstructured`.